internal/datastores/vvol_datastore_subresource.go

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2023 VMware, Inc.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package datastores
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vcf-sdk-go/models"
+	"strings"
+)
+
+// VvolDatastoreSchema this helper function extracts the VVOL Datastore schema, so that
+// it's made available for both Domain and Cluster creation.
+func VvolDatastoreSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"datastore_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Datastore name used for cluster creation",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"storage_container_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "UUID of the VASA storage container",
+				ValidateFunc: validation.IsUUID,
+			},
+			"storage_protocol_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Type of the VASA storage protocol. One among: ISCSI, NFS, FC",
+				ValidateFunc: validation.StringInSlice(
+					[]string{"ISCSI", "NFS", "FC"}, true),
```

Comment:
Why not...

```go
var vVolStorageProtocols = []string{
	"ISCSI",
	"NFS",
	"FC",
}
```

And then... 

```go
ValidateFunc: validation.StringInSlice(vVolStorageProtocols, false),
```

Reply:
Vars can be changed, []string cannot be a constant, so I have to create a fuction that returns the []string. I don't see the value...